nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -827,7 +827,7 @@ func (w *deploymentWatcher) getEval() *structs.Evaluation {
 	return &structs.Evaluation{
 		ID:           uuid.Generate(),
 		Namespace:    w.j.Namespace,
-		Priority:     w.j.Priority,
+		Priority:     w.d.EvalPriority,
```

Comment:
During a server upgrade it's possible we end up with deployments created on the previous version that are then "watched" on a leader that's on the new version. So I think when we call `getEval` on that new leader we could get a deployment with an eval priority of 0. Maybe we should have some belt-and-suspenders here where we pick up the job priority if the `EvalPriority` is 0?